### PR TITLE
Adding Java SideEffect doc

### PR DESCRIPTION
**Pull request by tsurdilo**
Signed-off-by: Tihomir Surdilovic <tihomir@temporal.io>

## What was changed:

Adding SideEffect doc for Java 
## Why?

It was missing
## Checklist

1. Closes issue: 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: sidebars.js [hunk 1]
```diff
@@ -163,6 +163,7 @@ module.exports = {
     'java/implementing-workflows',
     'java/starting-workflow-executions',
     'java/versioning',
+    'java/side-effect'
```
Comment (flossypurse) [inline sidebars.js:166]:
```suggestion
    'java/side-effect',
```
😉

**Review by flossypurse — CHANGES_REQUESTED** [review]
1. Can we put each sentence on a newline?
2. Make sure "Workflows", "Activities", etc.. are capitalized.

**Comment by tsurdilo** [comment]
@flossypurse updated, sorry

**Review by flossypurse — APPROVED** [review]

**Comment by flossypurse** [comment]
Excellent @tsurdilo - I think you made this so simple and clear.

**Pull request merged**